src/cmd/create.go

```diff
@@ -190,12 +207,46 @@ func createContainer(container, image, release string, showCommandToEnter bool)
 		return errors.New(errMsg)
 	}
 
-	pulled, err := pullImage(image, release)
-	if err != nil {
-		return err
-	}
-	if !pulled {
-		return nil
+	isPulled, image := isImagePulled(image, release)
+	if !isPulled {
+		if image == "" {
+			var builder strings.Builder
+			fmt.Fprintf(&builder, "image %s not found in local storage and known registries\n", image)
+			fmt.Fprintf(&builder, "Make sure the image URI is correct")
+
+			errMsg := builder.String()
+			return errors.New(errMsg)
+		}
+
+		domain := utils.ImageReferenceGetDomain(image)
```

Comment:
I was hoping that we wouldn't be significantly increasing the length of `createContainer`, and keep the details of authentication encapsulated within `pullImage`. :)

In terms of creating a container, whether we need to authenticate or not to pull an image, is an implementation detail.

Reply:
The context also did not fit into `pullImage` :). I'll shuffle the code around a bit.